src/app/auth/auth.component.ts

```diff
@@ -177,17 +173,22 @@ export class AuthComponent implements OnInit, OnDestroy {
    * @param username - The user to log in.
    */
   completeLoginProcess(username: string): void {
-    this.authService.changeAuthState(AuthState.AwaitingAuthentication);
-    this.phaseService.setPhaseOwners(this.currentSessionOrg, username);
-    this.userService.createUserModel(username).pipe(
-      flatMap(() => this.phaseService.sessionSetup()),
-      flatMap(() => this.githubEventService.setLatestChangeEvent()),
-    ).subscribe(() => {
-      this.handleAuthSuccess();
-    }, (error) => {
-      this.authService.changeAuthState(AuthState.NotAuthenticated);
-      this.errorHandlingService.handleError(error);
-      this.logger.info(`Completion of login process failed with an error: ${error}`);
+    const org = window.localStorage.getItem('org');
+    const dataRepo = window.localStorage.getItem('dataRepo');
+    this.githubService.storeOrganizationDetails(org, dataRepo);
+    this.phaseService.storeSessionData().subscribe(() => {
```

Comment:
Calling `phaseService.storeSessionData()` here creates a nested situation where we subscribe to one observable inside the subscription callback of another observable.

To avoid this confusing situation, let's store the `SessionData` object in local storage and create a new `setSessionData` method in phaseService. The `SessionData` would be saved before the redirect to GitHub, and restored after the redirect to CATcher - just like what you did with `githubService.storeOrganizationDetails(org, dataRepo);`.

What do you think  @AudreyFelicio ?

Reply:
@anubh-v 
Yes, this is a great idea!

Apologies for the nested observables confusion 😢 .
I was originally just trying to mimic `Promise.then` using the `subscribe` method as we need the `storeSessionData` to finish before doing the `flatMap(() => this.phaseService.sessionSetup())`. The problem indeed comes from the asynchronous nature of `storeSessionData`.

Just to confirm my understanding, the `setSessionData` method should be a normal synchronous method that fetches the `SessionData` from local storage and calls `phaseService.updateSessionParameters`, right?